### ISSUE TITLE
VisibilityEnabler2D forces Particle2D to emit

### ISSUE THREAD

**Issue by carvalho-paulo** [godotengine/godot#74827]
### Godot version

3.5.1 stable

### System information

Windows 10

### Issue description

If we have in a scene a Particle2D with Emitting off and a VisibilityEnabler2D with Pause Particles on, then the Particle2D will emit anyway.

### Steps to reproduce

1. Create a Node2D scene
2. Add Particle2D with Emitting off in the editor
3. Play the scene: OK, particles don't emit as expected
4. Add VisibilityEnabler2D with Pause Particles on in the editor
5. Play the scene: particles are emitting now

The expected is that particles don't emit in step 5.

### Minimal reproduction project

[VisibilityEnabler2D Particle2D Issue.zip](https://github.com/godotengine/godot/files/10951376/VisibilityEnabler2D.Particle2D.Issue.zip)

**Comment by lawnjelly** [comment]
Yes, it's rather naive at the moment and maybe only suitable for continuous emitters:
```
	if (enabler[ENABLER_PAUSE_PARTICLES]) {
		Particles2D *ps = Object::cast_to<Particles2D>(p_node);

		if (ps) {
			ps->set_emitting(p_enabled);
		}
	}
```
It probably needs another flag for this to work correctly as you would expect.
